metadata/metadata.go

```diff
@@ -1423,6 +1437,53 @@ func (resource *entityResource) Update(lookup ResourceLookup, updateRes Resource
 	return nil
 }
 
+type triggerResource struct {
+	serialized *pb.Trigger
+}
+
+func (resource *triggerResource) ID() ResourceID {
+	return ResourceID{
+		Name: resource.serialized.Name,
+		Type: TRIGGER,
+	}
+}
+
+func (resource *triggerResource) Schedule() string {
+	// TODO: Make sure it's returning the correct thing
+	return resource.serialized.GetScheduleTrigger().Schedule
+}
+
+func (resource *triggerResource) Dependencies(lookup ResourceLookup) (ResourceLookup, error) {
+	// TODO: Change later, do we need to return all the resources that use this trigger?
```

Comment:
the only current validation rule we have is we can't delete triggers that have resources currently, so if it's a performance issue, we only need to know that it has some resources not necessarily all of them.